src/state/swap/index.ts

```diff
@@ -85,6 +85,17 @@ interface InputEventHandlers {
 /** An integration hook called when the user approves a token, either through allowance or permit. */
 export type OnSwapApprove = () => void
 
+/** An integration hook called when a new quote is fetched. */
+export type OnSwapQuote = (
```

Comment:
do you think there's any benefit to making `event` fields optional? i'm thinking of situations where we need to change the API (adding/removing fields without disrupting integrators). maybe this only applies to when we add or remove later?

Reply:
I don't think there's a benefit - I don't think there's a large issue with changing this down the road.